src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -581,7 +581,11 @@ export class RunActiveFileInTerminalAction extends Action {
 			this.notificationService.warn(nls.localize('workbench.action.terminal.runActiveFile.noFile', 'Only files on disk can be run in the terminal'));
 			return TPromise.as(void 0);
 		}
-		instance.sendText(uri.fsPath, true);
+
+		const filePath = this.terminalService.isWslBashTerminal(instance) ?
+			uri.path.replace(/([A-Za-z]):/, (m, g) => `mnt/${g.toLocaleLowerCase()}`) :
+			uri.fsPath;
+		instance.sendText(`"${filePath}"`, true);
```

Comment:
It would be best if `TerminalInstance` knew if it was WSL, I think all this logic should move into something like `TerminalInstance.sendFileName(uri: Uri): void` which the action calls.